src/Microsoft.AspNet.Authentication.OAuth/OAuthAuthenticationOptions.cs

```diff
@@ -109,5 +101,12 @@ public string Caption
         /// Gets or sets the type used to secure data handled by the middleware.
         /// </summary>
         public ISecureDataFormat<AuthenticationProperties> StateDataFormat { get; set; }
+
+        /// <summary>
+        /// Defines whether access and refresh tokens should be stored in the <see cref="ClaimsPrincipal"/>
+        /// after a successful authentication. You can set this property to <see cref="false"/> to reduce
+        /// the size of the final authentication cookie.
+        /// </summary>
+        public bool SaveTokens { get; set; } = true;
```

Comment:
Maybe SaveTokensAsClaims or ExposeTokenClaims to be slightly more clear what this is doing and used for?  I don't think its super obvious what SaveTokens means by itself...